### PR TITLE
added a user-defined scheduler tutorial and fixed x-axis label cut-offs

### PR DESCRIPTION
**Pull request by kvyh**
This adds a tutorial for making a user-defined `Scheduler`.

### CODE REVIEW

File: docs/tutorials/scheduling.rst [hunk 1]
```diff
@@ -353,7 +356,236 @@ and run the priority scheduler again.
...
+are currently implemented. This example will walk through the steps for
+creating your own scheduler that will be compatible with the tools of
+the ``scheduling`` module.
+
+As you may have noticed above, the schedulers are assembled by making a
+call to the initializer of the class (e.g. `~astroplan.scheduling.PriorityScheduler`).
+Each of the schedulers is subclassed from the abstract `astroplan.scheduling.Scheduler`
+class, and our custom scheduler needs to be as well.
+
+For our scheduler, we will make one starts at the beginning and schedules
```
Comment (bmorris3) [inline docs/tutorials/scheduling.rst]: Grammar is funky in this sentence.

File: docs/tutorials/scheduling.rst [hunk 2]
```diff
@@ -353,7 +356,236 @@ and run the priority scheduler again.
...
+the first ``ObservingBlock`` that it finds that doesn't have a score of zero: a
```
Comment (bmorris3) [inline docs/tutorials/scheduling.rst]: Could you start at a higher level than this? First example: we are going to compute constraints, and some will be evaluated as non-boolean constraints, yielding scores. Then we we will look to maximize score... etc.

File: docs/tutorials/scheduling.rst [hunk 3]
```diff
@@ -353,7 +356,236 @@ and run the priority scheduler again.
...
+  the `~astroplan.Observer`, the `~astroplan.scheduling.Transitioner`, a ``gap_time``,
+  and a ``time_resolution`` for spacing during the creation of the schedule.
+
+* It also needs a ``_make_schedule`` to do the heavy lifting. This takes a list of
+  `~astroplan.scheduling.ObservingBlock` objects and a `~astroplan.scheduling.Schedule`
+  object to input them into. This method needs to be able to check whether a
+  block can be scheduled in a given spot, and be able to insert it into the
+  schedule once a suitable spot has been found.
+
+Here's the ``MinimumBarScheduler`` implementation::
```
Comment (bmorris3) [inline docs/tutorials/scheduling.rst]: What is the _bar_ in `MinimumBarScheduler`? Maybe `SimpleScheduler` would be good?

File: docs/tutorials/scheduling.rst (hunk 2 above)
Comment (kvyh) [inline docs/tutorials/scheduling.rst]: I added some explanation of score/constraints, but the example scheduler doesn't actually try to maximize score. In a future commit I have added a few comments that talk about how a different scheduler might try to maximize the score and how scores are evaluated.

File: docs/tutorials/scheduling.rst [hunk 4]
```diff
@@ -353,7 +358,267 @@ and run the priority scheduler again.
...
+    >>> from astropy.time import Time
+
+    >>> apo = Observer.at_site('apo')
+    >>> Deneb = FixedTarget.from_name('Deneb')
+    >>> M13 = FixedTarget.from_name('M13')
+    >>> blocks = [ObservingBlock(Deneb, 20*u.minute, 0)]
+    >>> blocks.append(ObservingBlock(M13, 20*u.minute, 0))
+
+    >>> # this will produce 0 second transitions
+    >>> transitioner = Transitioner()
```
Comment (bmorris3) [inline docs/tutorials/scheduling.rst]: If we used a more realistic `Transitioner`, would that make the outputs more difficult to interpret? I think that's a bit too unrealistic.

**Comment by kvyh** [comment]
I made the `transitioner` more realistic and cut the lines so that they would fit with nearly no scrolling when built. I also added some clarity in the inputs for `from_duration`.

File: docs/tutorials/scheduling.rst [hunk 5]
```diff
@@ -353,7 +379,278 @@ and run the priority scheduler again.
...
+multiplying the scores returned by all of the constraints for that
+target and time.
+
+If you wish to use a different method of score evaluation, I would
+suggest adding a new method to the ``Scorer``. Use the general
+framework of the ``create_score_array`` method, but change how it
+combines the scores from the separate constraints (e.g. add the
+reciprocals of the scores together and then use smaller values
+as better). If you create a useful method, consider submitting it
+to http://github.com/astropy.astroplan so that other's will be able
```
Comment (bmorris3) [inline docs/tutorials/scheduling.rst]: Would you make this an embedded link? Here's the format: 
```
`consider submitting a pull request <http://github.com/astropy/astroplan/pulls>_` 
```

**Comment by bmorris3** [comment]
Would you also make inline comments in the example code? For example: 

```python

>>> from astroplan.scheduling import SequentialScheduler
>>> from astroplan.scheduling import Schedule

>>> # Initialize the sequential scheduler with the constraints and transitioner
>>> seq_scheduler = SequentialScheduler(constraints = global_constraints,
...                                     observer = apo,
...                                     transitioner = transitioner)

>>> # Initialize a Schedule object, to fill with our new schedule
>>> sequential_schedule = Schedule(noon_before, noon_after)

>>> # Call the scheduler on the observing blocks, put the scheduled blocks into the schedule
>>> seq_scheduler(blocks, sequential_schedule)
```

File: docs/tutorials/scheduling.rst [hunk 6]
```diff
@@ -288,6 +307,7 @@ and run the priority scheduler again.
 .. code-block:: python
 
     >>> alf_cent = FixedTarget.from_name('Alpha Centauri A')
```
Comment (bmorris3) [inline docs/tutorials/scheduling.rst]: While `alf_cent` could be technically correct spelling in some circles, would you change it everywhere to `alpha_cen`, which is more readible?

File: docs/tutorials/scheduling.rst [hunk 7]
```diff
@@ -288,6 +307,7 @@ and run the priority scheduler again.
...
+    >>> # ObservingBlocks can also be called with a simple (target, duration, priority)
```
Comment (bmorris3) [inline docs/tutorials/scheduling.rst]: This line would be clearer if you stated: 

```
# ObservingBlocks can also be called arguments: target, duration, priority
```

**Comment by kvyh** [comment]
I have added the edits and a few additional in-line comments.

**Comment by bmorris3** [comment]
I think we can incrementally improve this tutorial with user feedback in the near future, but I will merge this now so we can move on to v0.2 ASAP.

**Pull request merged**